### ISSUE TITLE
Can I start scroll animation when the specific portion of div is in viewport

### ISSUE THREAD

**Issue by Umair45** [scrollreveal/scrollreveal#501]
Hi,

First of all thanks for this library. I've one thing to ask.

I need to start animation only once a specific link is in the viewport in my banner, I've added my banner here. please open the following link and scroll down to see the banner area and animation:

http://kitndkaboodle.com/banner/

I've applied animations on the statistics and bottom link and its works fine as required. But I want only to start animation when link is visible otherwise not. for the reference please see the following screenshot i've added. Red line means I want to start animation when the link is visible.

![section](https://user-images.githubusercontent.com/16554703/61283917-58bb7c00-a7d7-11e9-87c1-035c027ddcc5.png)

### Environment

* Operating System: Windows
* Browser Version: Version 75.0.3770.100 (Official Build) (64-bit)
* ScrollReveal Version: ScrollReveal v4.0.5

I would be highly appreciate the help on this asap. As I need to fix this thing.

Best Regards,
Umair

**Comment by jlmakes** [comment]
There isn’t first-class support separating visibility checks from animation triggers. You’ll have to devise a workaround to achieve this behavior with the current version of ScrollReveal (4.x).

Here’s an idea you can try. If you manually hide the statistics elements, you can reveal them in the `afterReveal` callback of a `reveal()` call on the link element. For example:

```
// pseudo code

otherElements.style.visibility = 'hidden'

reveal(link, {
  afterReveal: () => {
    reveal(otherElements)
  }
})
```

Something like this could work pretty well for a single reveal (without resets). But keep in mind that if JavaScript were disabled, or if ScrollReveal were unsupported, your elements would remain hidden and never be revealed.

**Comment by jlmakes** [comment]
There isn’t first-class support separating visibility checks from animation triggers. You’ll have to devise a workaround to achieve this behavior with the current version of ScrollReveal (4.x).

Here’s an idea you can try. If you manually hide the statistics elements, you can reveal them in the `afterReveal` callback of a `reveal()` call on the link element. For example:

```
// pseudo code

otherElements.style.visibility = 'hidden'

reveal(link, {
  afterReveal: () => {
    reveal(otherElements)
  }
})
```

Something like this could work pretty well for a single reveal (without `option.reset`). But keep in mind that if JavaScript were disabled, or if ScrollReveal were unsupported, your elements would remain hidden and never be revealed. Your implementation would probably want to account for that.

P.S. ScrollReveal has an undocumented static method `ScrollReveal.isSupported()` that you can use to check if the current browser is supported.

**Comment by Umair45** [comment]
Thanks a million for getting into my issue and reply.
But my query was that this reveal animation should only happen when anchor above the red line is in the viewport. Right now, it starts animation once the starting point of my banner is in the viewport and its not taking any notice that wheather elements in the banner are in the viewport or not. Its just get started the animation.
but I found solution for this from the Library options. Following link helped me.

https://scrollrevealjs.org/api/view-factor.html

I just added the "viewFactor: 1.0" attribute and my problem fixed. So, now animation not starts now until my statistics and link is in the viewport

Thanks,
